src/include/common/object_pool.h

```diff
@@ -33,7 +35,7 @@ struct DefaultConstructorAllocator {
  *         handed out multiple times before that happens.
  */
 template <typename T, class Allocator = ByteAllocator<T>>
-class ObjectPool {
+class ObjectPool : public statistics::PerformanceCounters<ObjectPool<T, Allocator>> {
```

Comment:
We can choose the counter as a member or inheritance. But if choose a member or private inheritance, we need a print method in measured class like `ObjectPool` instead of `PrintPerformanceCounters()` method in the performance counter. Anyway, I will read this discussion.

Reply:
I changed the code to be injected as above link.